### PR TITLE
Updating trades whenever the order's executed amounts change

### LINKED ISSUE #407: Etherscan link is not shown automatically
**Describe the bug**
User needs to refresh to see the etherscan link

**How to Reproduce**

1. Trade in cowswap
2. Open explorer while the order is OPEN
3. Wait until the trade is executed
4. Verify if the link is displayed automatically

**Expected behavior**
User sees the link to etherscan without needing to refresh

**Additional context**
Found by our QA expert @MareenG https://github.com/gnosis/gp-ui/pull/404#pullrequestreview-645573171

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Closes #407

Updating tx hash whenever trade happens

# Testing

1. Using a local CowSwap version, or cowswap.dev.gnosisdev.com (Staging/Prod version won't work), place an order
2. Open the Explorer link and copy the order id
3. Open this PR's Explorer app and load the order id
4. Observe the order details page until a match happens
- [ ] When the trade happens, the tx hash field should auto-populate without the need for a page refresh

### CODE REVIEW

File: src/hooks/useOperatorTrades.ts [hunk 1]
```diff
@@ -97,8 +97,10 @@ export function useOrderTrades(order: Order | null): Result {
 
     fetchTrades(networkId, order)
     // Depending on order UID to avoid re-fetching when obj changes but ID remains the same
+    // Depending on `executedBuy/SellAmount`s string to force a refetch when there are new trades
+    // using the string version because hooks are bad at detecting Object changes
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [fetchTrades, networkId, order?.uid])
+  }, [fetchTrades, networkId, order?.uid, order?.executedSellAmount.toString(), order?.executedBuyAmount.toString()])
```
Comment (alfetopito) [inline src/hooks/useOperatorTrades.ts:103]: Of course this is a hack ;)

Hope the comment is clear enough.
Can provide further details if needed.
But in that case I should also update the comment.

Comment (W3stside) [reply src/hooks/useOperatorTrades.ts:103]: This shouldn't really be the case should it? I feel we'd have seen this in other places too. But if you say it works..

**Review by W3stside — APPROVED** [review]

File: src/hooks/useOperatorTrades.ts (hunk 1 above)
Comment (anxolin) [reply src/hooks/useOperatorTrades.ts:103]: yikes! but fair enough, and with the comments sounds alright.

**Review by anxolin — COMMENTED** [review]
APPROVE

**Pull request merged**